### DISCUSSION TITLE
Version 2.1.2

### DISCUSSION THREAD

**Discussion by rainepretorius** [rainepretorius/olarm-ha-integration#51] [Announcements]
**Bugfixes:**

- Workaround for Too Many Requests
- Fixed Device select not working from UI.

<hr /><em>This discussion was created from the release <a href='https://github.com/rainepretorius/olarm-ha-integration/releases/tag/version2.1.2'>Version 2.1.2</a>.</em>

**Comment by rwoeke** [comment]
Not allowing to add the integration after this update: See log

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/olarm_sensors/__init__.py", line 46, in async_setup_entry
    await update_listener(hass, config_entry)
  File "/config/custom_components/olarm_sensors/__init__.py", line 173, in update_listener
    if config_entry.options[CONF_SCAN_INTERVAL] < 30 or config_entry.data[CONF_SCAN_INTERVAL]:
KeyError: 'scan_interval'

**Reply by Chris-777** [reply]
Exactly the same for me, see log:

Error setting up entry Olarm Sensors for olarm_sensors

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/olarm_sensors/__init__.py", line 46, in async_setup_entry
    await update_listener(hass, config_entry)
  File "/config/custom_components/olarm_sensors/__init__.py", line 173, in update_listener
    if config_entry.options[CONF_SCAN_INTERVAL] < 30 or config_entry.data[CONF_SCAN_INTERVAL]:
KeyError: 'scan_interval'

**Reply by rainepretorius** [reply]
> Not allowing to add the integration after this update: See log
> 
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
>     result = await component.async_setup_entry(hass, self)
>   File "/config/custom_components/olarm_sensors/__init__.py", line 46, in async_setup_entry
>     await update_listener(hass, config_entry)
>   File "/config/custom_components/olarm_sensors/__init__.py", line 173, in update_listener
>     if config_entry.options[CONF_SCAN_INTERVAL] < 30 or config_entry.data[CONF_SCAN_INTERVAL]:
> KeyError: 'scan_interval'

Damn. Will see what can cause this issue when I get back home tonight.

**Comment by rainepretorius** [comment]
> Not allowing to add the integration after this update: See log
> 
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
>     result = await component.async_setup_entry(hass, self)
>   File "/config/custom_components/olarm_sensors/__init__.py", line 46, in async_setup_entry
>     await update_listener(hass, config_entry)
>   File "/config/custom_components/olarm_sensors/__init__.py", line 173, in update_listener
>     if config_entry.options[CONF_SCAN_INTERVAL] < 30 or config_entry.data[CONF_SCAN_INTERVAL]:
> KeyError: 'scan_interval'

I just quickly tried to implement a fix and did not test. My appologies for the inconvenience.
